### ISSUE TITLE
False negative - FURB188 removesuffix with string literal

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#16231]
### Description

```python
a = "sjdfaskldjfakljklfoo"
if a.endswith("foo"):
    a = a[: -len("foo")]
print(a)
```
should be transformed to
```python
a = "sjdfaskldjfakljklfoo"
a = a.removesuffx("foo")
print(a)
```
when FURB188 is enabled.  Other similar trivial cases are caught as seen by the changes I made in PyTorch for https://github.com/pytorch/pytorch/pull/146997 so I assume it's an error in the rule logic where some corner case is not properly handled.

Ruff version is `ruff 0.9.6`
Command is `ruff check --select=FURB188 --target-version py39 test_FURB188.py`

Interestingly:
```python
a = "sjdfaskldjfakljklfoo"
suffix = "foo"
if a.endswith(suffix):
    a = a[: -len(suffix)]
print(a)
```
is properly flagged currently, so it seems to have issue with string literals, and only for suffixes accessed in this pattern interestingly enough.
